http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala

```diff
@@ -306,411 +247,267 @@ object RuntimeServiceDbQueries {
-                  C.`hostIp`,
-                  C.`internalId`,
-                  C.`workspaceId`,
-                  C.`cloudProvider`,
-                  C.`id`,
-                  C.`runtimeName`,
-                  C.`kernelFoundBusyDate`,
-                  C.`creator`,
-                  C.`proxyHostName`,
-                  C.`runtimeConfigId`
-                from
-                  `CLUSTER` AS C
-                #${whereFilterClusters}
-              ) AS FILTERED_CLUSTER
-              left join `LABEL` L on (L.`resourceId` = FILTERED_CLUSTER.id) and (L.`resourceType` = 'runtime')
-              #${whereFilterLabels}
-          ) AS LABEL_FILTERED
-          inner join `RUNTIME_CONFIG` RG on LABEL_FILTERED.runtimeConfigId = RG.`id`
-          left join `CLUSTER_PATCH` CP on LABEL_FILTERED.id = CP.`clusterId`
-          GROUP BY LABEL_FILTERED.id""".stripMargin
-
-    sqlStatement.as[ListRuntimeResponse2]
-  }
-
-  private def listRuntimesHelper(labelMap: LabelMap,
-                                 excludeStatuses: List[RuntimeStatus],
-                                 creatorOnly: Option[WorkbenchEmail],
-                                 workspaceId: Option[WorkspaceId],
-                                 cloudContextOrCloudProvider: Option[Either[CloudContext, CloudProvider]]
-  ): DBIO[Vector[ListRuntimeResponse2]] = {
-    val runtimeQueryFilteredByCreator = creatorOnly match {
-      case Some(creator) => List(s"C.`creator` = '${creator.value}'")
-      case None          => List.empty
-    }
-    val runtimeQueryFilteredByDeletion = excludeStatuses.map(s => s"C.`status` != '${s.toString}'")
-
-    val runtimeQueryFilteredByWorkspace = workspaceId match {
-      case Some(wid) =>
-        List(s"C.`workspaceId` = '${wid.value.toString}'")
-      case None => List.empty
-    }
-
-    val runtimeQueryFilteredByCloud = cloudContextOrCloudProvider match {
-      case Some(Left(CloudContext.Gcp(gp)))     => List(s"C.`cloudContext` = '${gp.value}'")
-      case Some(Left(CloudContext.Azure(actx))) => List(s"C.`cloudContext` = '${actx.asString}'")
-      case Some(Right(cloudProvider))           => List(s"C.`cloudProvider` = '${cloudProvider.asString}'")
-      case None                                 => List.empty
-    }
-
-    val clusterFilters =
-      (runtimeQueryFilteredByCreator ++ runtimeQueryFilteredByDeletion ++ runtimeQueryFilteredByWorkspace ++ runtimeQueryFilteredByCloud)
-        .mkString(" AND ")
+    val runtimesFiltered = labelMap
+      .foldLeft(runtimesFilteredSimple) {
+        case (someRuntimeQuery: Query[ClusterTable, ClusterRecord, Seq], (key: String, value: String)) =>
+          for {
+            (runtime, _) <- someRuntimeQuery join labelQuery on ((r, l) =>
+              l.resourceId === r.id &&
+                l.resourceType === LabelResourceType.runtime &&
+                l.key === key &&
+                l.value === value
+            )
+          } yield runtime: ClusterTable
+      }
```

Comment:
Label filters! All of the given label terms must be true, so we do a series of INNER JOINs, discarding the label side of the result.

Reply:
This comment is outdated now - the label query is now built up mostly identically to the old style, but included in the innermost query against runtimes.